From 808e7d6b0f8dd81e2d99f85c9ef94006040719dc Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Tue, 14 Nov 2017 11:45:42 -0500
Subject: [PATCH 97/97] crypto: ks2: add KNAV_QUEUE-SHARED to tx queue

The keystone side band crypto driver and inflow driver share the same
tx queue to push packets to the SA accelerator. This commit adds the
KNAV_QUEUE-SHARED flag for the queue. Without the flag the driver
cannot open the queue as it is already opened by the inflow driver.

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
---
 drivers/crypto/keystone-sa.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/crypto/keystone-sa.c b/drivers/crypto/keystone-sa.c
index 566b383..72c2140 100644
--- a/drivers/crypto/keystone-sa.c
+++ b/drivers/crypto/keystone-sa.c
@@ -347,7 +347,8 @@ static int sa_setup_resources(struct keystone_crypto_data *dev_data)
 
 	snprintf(name, sizeof(name), "tx-subm-q-%s", dev_name(dev));
 	dev_data->tx_submit_q = knav_queue_open(name,
-						dev_data->tx_submit_qid, 0);
+						dev_data->tx_submit_qid,
+						KNAV_QUEUE_SHARED);
 	if (IS_ERR(dev_data->tx_submit_q)) {
 		ret = PTR_ERR(dev_data->tx_submit_q);
 		dev_err(dev, "Could not open \"%s\": %d\n", name, ret);
-- 
1.9.1

